client/errors.go

```diff
@@ -18,24 +18,59 @@ type DetailedError struct {
 	// Title is a human-readable summary that explains the type of the problem.
 	Title string `json:"title,omitempty"`
 	// Details is an array of structured objects that give details about the error.
-	Details []struct {
-		Code        string `json:"code"`
-		Description string `json:"description"`
-		Field       string `json:"field"`
-	} `json:"type_detail,omitempty"`
+	Details []TypeDetail `json:"type_detail,omitempty"`
+}
+
+type TypeDetail struct {
```

Comment:
Maaaybe worth calling this `ErrorTypeDetail` or some thing? Naming is hard 🤷🏻

Reply:
done!